src/MHTimer.js

```diff
@@ -496,12 +497,21 @@ function parseUserMessage(message) {
 
         // Display or update the user's reminders.
         case 'remind':
-            // TODO: redirect responses to PM.
-            if (!tokens.length || !reminderRequest.area)
-                listRemind(message);
-            else
-                addRemind(reminderRequest, message);
-            break;
+            {
+                let dataCatcher;
+                // TODO: redirect responses to PM.
+                if (!tokens.length || !reminderRequest.area)
+                    listRemind(message).then(function(result){dataCatcher = result});
+                else
+                    addRemind(reminderRequest, message).then(function(result){dataCatcher = result});
+                
+                if(dataCatcher.getDm && !dataCatcher.getSuccess) {
```

Comment:
The reason I suggested wrapping the calls to listRemind, addRemind, etc with another function is to avoid what is happening here: 

1)  `dataCatcher` is declared and assigned no value.
2) the asynchronous function listRemind or addRemind is invoked.
3) this line is executed after the method in 2) schedules something for the next event loop (i.e. does something asyc). At this time, `dataCatcher` is still `undefined`, so this line will throw "TypeError: cannot read property 'getDm' of 'undefined'"
3) at some later point, the method from 2) will resolve
4) some time after that, the respective `then` handler is invoked, setting a value for `dataCatcher`